ttnn/cpp/ttnn/operations/experimental/transformer/rotary_embedding_llama/rotary_embedding_llama.cpp

```diff
@@ -20,7 +20,9 @@ Tensor RotaryEmbeddingLlamaOperation::invoke(
     operation::launch_op(
         [memory_config, compute_kernel_config] (const std::vector<Tensor>& input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors, const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
             auto& input_tensor = input_tensors.at(0);
-            uint32_t seq_len = input_tensor.get_legacy_shape()[-2];
+            bool is_sharded = input_tensor.memory_config().memory_layout == TensorMemoryLayout::HEIGHT_SHARDED; // TODO: Check sharding direction
```

Comment:
This is very misleading since sharding can be height, width, or block sharded. Definitely do not infer mode based on whether something is height sharded or not. Does the op not support the other sharding?

Also, you don't need to propagate a flag like this in configs. Just get it where you need it.

Reply:
yeah the op only supports height sharded. I added better validation code/messages for that.

also got rid of the flag from the config and rather derived it when needed.